### ISSUE TITLE
Potential defect in `stm32f4_probe()`

### ISSUE THREAD

**Issue by stoyan-shopov** [blackmagic-debug/blackmagic#1328]
https://github.com/blackmagic-debug/blackmagic/blob/5ac4378b77095326e7f096c0a440f1060200a63b/src/target/stm32f4.c#L234

Hi, team
I suspect there could be a defect with this code, in file `stm32f4.c`, function `stm32f4_probe()`, line 234:
```
		t->driver = stm32f4_get_chip_name(t->part_id);
		t->part_id = device_id;
```
I think these two lines of code should be interchanged, but I may be wrong.

I am getting an **STM32F411** device reported ig **gdb** as **STM32F2 M4**.
I happen to have some probe with old firmware, and that one reports in **gdb** **STM32F40x M4**

Feel free to close this issue if inappropriate

Thanks

**Comment by perigoso** [comment]
Hi, you are correct in that that looks wrong, but it might be more appropriate to rewrite `t->driver = stm32f4_get_chip_name(t->part_id);` as `t->driver = stm32f4_get_chip_name(device_id);`

**Comment by perigoso** [comment]
can you try #1329?

**Comment by stoyan-shopov** [comment]
Thank you, @perigoso

I confirm that your change works

I just tried your change on an **STM32F411 blackpill** as a BMP host.
It successfully recognizes another **STM32F411 blackpill**:
```
(gdb) mon s
Available Targets:
No. Att Driver
 1      STM32F411 M4
```
I also tried to connect to an old **blackpill**, with an **STM32F401** part:
```
(gdb) mon s
Available Targets:
No. Att Driver
 1      STM32F401C M4
```

(For the record, with an **STM32F411 blackpill**, pressing the reset button takes the board into DFU mode - so, in order to actually reset the probe, power needs to be cycled. This has nothing to do with this change, just mentioning. I will have a look at the probe initialization code, maybe there is some discrepancy with the probes I use.)

Thanks a lot!
